lib/auth.js

```diff
@@ -142,6 +142,33 @@ bedrock.events.on('bedrock-session-http.session.get', function(req, session) {
 
 bedrock.events.on('bedrock-views.vars.get', addViewVars);
 
+bedrock.events.on('bedrock-identity.insert', function(options, callback) {
+  // add IDP identity defaults
+  var identity = options.identity;
+
+  var defaults = bedrock.util.clone(config.idp.defaults);
+  var clonedIdentity = bedrock.util.extend(
+    {}, defaults.identity, bedrock.util.clone(identity));
+  // Must mutate original identity object, so assign all props back over
+  for(var key in clonedIdentity) {
+    identity[key] = clonedIdentity[key];
+  }
+  // create identity ID from slug if not present
+  if(!('id' in identity)) {
+    identity.id = api.createIdentityId(identity.sysSlug);
+  }
+
+  // if using a 'did:', add resource role for local id
+  if(identity.id.indexOf('did:') === 0) {
+    identity.sysResourceRole = identity.sysResourceRole || [];
+    identity.sysResourceRole.push({
+      sysRole: 'identity.registered',
+      resource: [api.createIdentityId(identity.sysSlug)]
```

Comment:
> Also, nothing up to this point is enforcing a sysSlug field to be present.

Well, that's because it was an explicit call before, so an assumption was made that there was appropriate validation before hand. However, now we've switched things around so that this code can get called when modules that know nothing about the idp call `brIdentity.insert`. So the model is a bit different. To mitigate this, I recommend that we check for `sysSlug` up front ... and return early by doing *nothing* if it's not set. We can use it as a flag for triggering idp behavior off/on -- and the slug is the main reason for the extra behavior anyway.